LiteCore/Storage/SQLiteKeyStore.cc

```diff
@@ -259,13 +259,16 @@ namespace litecore {
 
     string SQLiteKeyStore::transformCollectionName(const string& name, bool mangle) {
         ostringstream ss;
-        for (uint8_t c : name) {
+        const char* name_cstr = name.c_str();
+        while(char c = *name_cstr++) {
+            if (c == '\\') {
+                continue;
```

Comment:
Is it not idempotent? Applying it to "Table" twice would generate R"(\\Table)" ?  (I noticed this change does not make it non-idempotent.

Reply:
I did this on purpose because

A. We have no need to doubly encode things since backslash is not a valid character for the user.
B. There are so many places that implicitly pass through this area now I felt it was safer to leave the name as-is if it was already processed.

Reply:
So, your assumption here is, if we hit the above "continue", the argument mangle must be false. Assertion here?

Reply:
Not quite.  If mangle is true, the backslash is skipped, but then a capital letter is encountered so it is re-added

Reply:
Oh, right. My bad. De-mangle; if mangle then mangle. Perfect.